### PR TITLE
PXB-2489 Flatten the table of contents (8.0)

### PR DESCRIPTION
**Pull request by patrickbirch**
M  source/backup_scenarios/compressed_backup.rst
M  source/glossary.rst
D  source/how-tos.rst
M  source/how_xtrabackup_works.rst
D  source/howtos/permissions.rst
M  source/index.rst
D  source/installation.rst
A  source/installation/binary-tarball.rst
A  source/prerequisites/comparison.rst
A  source/prerequisites/configuring.rst
A  source/prerequisites/connections.rst
A  source/prerequisites/permissions.rst
D  source/using_xtrabackup/comparison.rst
D  source/using_xtrabackup/configuring.rst
D  source/using_xtrabackup/privileges.rst
M  source/xbcloud/xbcloud_azure.rst

### CODE REVIEW

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 1]
```diff
@@ -4,73 +4,106 @@
...
+   .opt
+     |MySQL| stores options of a database (like charset) in a file with a
+     :file:`.opt` extension in the database directory.
+
+   .par
+     Each partitioned table has .par file which contains metadata about the
+     partitions.
+
+   .TRG
+     File containing the Triggers associated to a table, for example
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: The file that contains the Triggers associated WITH a table, for example,

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 2]
```diff
@@ -4,73 +4,106 @@
...
+     `:file:`mytable.TRG`. With the :term:`.TRN` file, they represent all the
+     Trigger definitions.
+
+   .TRN
+     File containing the Triggers' Names associated to a table, for example
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: The file that contains the Triggers' Names associated WITH a table, for example,

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 3]
```diff
@@ -4,73 +4,106 @@
...
+  
+   datadir
+    The directory in which the database server stores its data files. Most Linux
+    distribution use `/var/lib/mysql` by default.
+
+   full backup
+     A backup that contains the complete source data from an instance. 
+
+   ibdata
+     Default prefix for tablespace files, for example :file:`ibdata1` is a 10MB
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: , for example,

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 4]
```diff
@@ -90,121 +123,74 @@
...
 
-   XtraDB
-     *Percona XtraDB* is an enhanced version of the InnoDB storage engine,
-     designed to better scale on modern hardware, and including a variety of
-     other features useful in high performance environments. It is fully
-     backwards compatible, and so can be used as a drop-in replacement for
-     standard InnoDB. More information `here
-     <https://www.percona.com/doc/percona-server/5.6/percona_xtradb.html>`_.
+   LSN
+     Each InnoDB page (usually 16kb in size) contains a log sequence number, or
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: a log sequence number or LSN

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 5]
```diff
@@ -90,121 +123,74 @@
...
+     page's LSN shows how recently it was changed.
 
    my.cnf
-     This file refers to the database server's main configuration file. Most
+     The database server's main configuration file. Most
      Linux distributions place it as :file:`/etc/mysql/my.cnf` or
      :file:`/etc/my.cnf`, but the location and name depends on the particular
      installation. Note that this is not the only way of configuring the
-     server, some systems does not have one even and rely on the command
+     server, some systems does not have one and rely on the command
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: some systems DO not have

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 6]
```diff
@@ -90,121 +123,74 @@
...
+     Previous default storage engine for |MySQL| for versions prior to 5.5. It
+     doesn't fully support transactions but in some scenarios may be faster
+     than :term:`InnoDB`. Each table is stored on disk in 3 files:
+     :term:`.frm`, :term:`.MYD`, :term:`.MYI`.
+  
+   physical backup
+     A backup that copies the data files.
+
+   point in time recovery
+     This method allows data to be restored into the state it was in any selected point of time.
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: This method restores the data into the state it was at any selected point in time.

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 7]
```diff
@@ -90,121 +123,74 @@
...
-   .TRG
-     File containing the Triggers associated to a table, e.g.
-     `:file:`mytable.TRG`. With the :term:`.TRN` file, they represent all the
-     Trigger definitions.
-
-   .TRN
-     File containing the Triggers' Names associated to a table, e.g.
-     `:file:`mytable.TRN`. With the :term:`.TRG` file, they represent all the
-     Trigger definitions.
+     To support simultaneous compression and streaming, |Percona XtraBackup| uses the xbstream format. For more information see :option:`--stream`
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: For more information, see

File: storage/innobase/xtrabackup/doc/source/glossary.rst [hunk 8]
```diff
@@ -90,121 +123,74 @@
...
+     To support simultaneous compression and streaming, |Percona XtraBackup| uses the xbstream format. For more information see :option:`--stream` 
 
-   .CSM
-     Each table with the :program:`CSV Storage Engine` has ``.CSM`` file which
-     contains the metadata of it.
+   XtraDB
+     *Percona XtraDB* is an enhanced version of the InnoDB storage engine,
+     designed to better scale on modern hardware. *Percona XtraDB* includes a variety of
+     other features useful in high performance environments. It is fully
+     backwards compatible, and so can be used as a drop-in replacement for
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/glossary.rst]: backward compatible

File: storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+See the :ref:`option and variable reference
+<xbk_option_reference>` for details on all of the configuration options.
+
+The *Percona XtraBackup* binary reads the ``[mysqld]`` and ``[xtrabackup]`` sections from any configuration files, in that order. This order allows the binary to read options from your existing |MySQL| installation, such as the `datadir` or certain *InnoDB* options. Specify any options to be overridden in the ``[xtrabackup]`` section. This section takes precedence since it is read later.
+
+The *Percona XtraBackup* binary does not accept exactly the same syntax in the
+`my.cnf` file as the `mysqld` server binary. For historical
+reasons, the `mysqld` server binary accepts parameters with a
+``--set-variable=<variable>=<value>`` syntax, which *Percona XtraBackup* does not understand. If your `my.cnf` file has these configuration directives rewrite them in the ``--variable=value`` syntax.
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst]: directives, rewrite them

File: storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+You are not required to put any options in your `my.cnf`. You can simply specify the options on the command-line. 
+
+The ``target_dir`` option may be convenient to put in the ``[xtrabackup]`` section
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst]: I would change to:
You can put the ``target_dir`` option into the ``[xtrabackup]`` section of your `my.cnf` file.

File: storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+---------------------------------------
+
+The *Percona XtraBackup* tool requires no special configuration on most systems.
+However, the storage where the ``--target-dir`` is located
+must behave properly when ``fsync()`` is called. In particular, we have noticed
+that NFS volumes not mounted with the ``sync`` option might not really sync the
+data. As a result, if you back up to an NFS volume mounted with the async
+option, and then try to prepare the backup from a different server that also
+mounts that volume, the data might appear to be corrupt. Use the
+``sync`` mount option to avoid this problem.
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst]: problem -> issue ?

File: storage/innobase/xtrabackup/doc/source/prerequisites/connections.rst [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+             TCP/IP.
+===========  ==================================================================
+
+These options are passed to the :command:`mysql` child process without
+alteration, see ``mysql --help`` for details.
+
+.. note::
+
+   In case of multiple server instances, the correct connection parameters
+   (port, socket, host) must be specified in order for *Percona XtraBackup* to talk to
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/connections.rst]: in order for *Percona XtraBackup* to talk -> for *Percona XtraBackup* to talk

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 13]
```diff
@@ -0,0 +1,134 @@
+
+.. _pxb.privilege:
+
+
+Permissions and Privileges Needed
+================================================================================
+
+For the purposes of this manual, *permissions* refers to the ability of a user to access and perform changes on the relevant parts of the host's filesystem, such as starting or stopping services, or installing software.
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: *permissions* refer

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+------------------------------
+
+The database user needs the following privileges on the tables or databases to be backed up:
+
+* ``RELOAD`` and ``LOCK TABLES`` (unless the `--no-lock <--no-lock>`
+  option is specified) in order to run :mysql:`FLUSH TABLES WITH READ LOCK` and
+  :mysql:`FLUSH ENGINE LOGS` prior to start copying the files, and requires this
+  privilege when `Backup Locks
+  <http://www.percona.com/doc/percona-server/8.0/management/backup_locks.html>`_
+  are used
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: I would paraphrase:
``RELOAD`` and ``LOCK TABLES`` privileges (unless the `--no-lock <--no-lock>`
  option is specified) — to run :mysql:`FLUSH TABLES WITH READ LOCK` and
  :mysql:`FLUSH ENGINE LOGS` before starting copying the files. These privileges are also needed  when `Backup Locks
  <http://www.percona.com/doc/percona-server/8.0/management/backup_locks.html>`_
  are used.

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+Using ``sudo`` in ``/etc/init.d/``, ``/etc/rc.d/`` or ``/sbin/service`` script provides the ability to start and stop services.
+
+Also, if you can execute the package manager of your distribution, you can install or remove software with it. If not, having ``rwx`` permission over a directory will let you do a local installation of software by compiling it there. This is a typical situation in many companies.
+
+There are other ways for managing permissions, such as using *PolicyKit*, *Extended ACLs* or *SELinux*, which may be preventing or allowing your access. You should check them in that case.
+
+Privileges Needed
+------------------------------
+
+The database user needs the following privileges on the tables or databases to be backed up:
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: The database user needs the following privileges to back up the tables or databases:

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+
+* ``RELOAD`` and ``LOCK TABLES`` (unless the `--no-lock <--no-lock>`
+  option is specified) in order to run :mysql:`FLUSH TABLES WITH READ LOCK` and
+  :mysql:`FLUSH ENGINE LOGS` prior to start copying the files, and requires this
+  privilege when `Backup Locks
+  <http://www.percona.com/doc/percona-server/8.0/management/backup_locks.html>`_
+  are used
+
+* ``BACKUP_ADMIN`` privilege is needed to query the
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: ``BACKUP_ADMIN`` privilege — to query

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+  performance_schema.log_status table, and run :mysql:`LOCK INSTANCE FOR BACKUP`,
+  :mysql:`LOCK BINLOG FOR BACKUP`, or :mysql:`LOCK TABLES FOR BACKUP`.
+
+* ``REPLICATION CLIENT`` in order to obtain the binary log position,
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: ``REPLICATION CLIENT`` privilege — to obtain

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+
+* ``CREATE TABLESPACE`` in order to import tables (see :ref:`pxb.xtrabackup.table.restoring`),
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: ``CREATE TABLESPACE`` privilege — to import tables

I suggest editing the following privileges the same way.

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+
+* ``SELECT`` privilege in order to use
+  :option:`--incremental-history-name` or
+  :option:`--incremental-history-uuid` in order for the feature
+  to look up the ``innodb_to_lsn`` values in the
+  :ref:`PERCONA_SCHEMA.xtrabackup_history <xtrabackup_history>` table.
+
+* ``SELECT`` privilege on the `keyring_component_status table <https://dev.mysql.com/doc/refman/8.0/en/performance-schema-keyring-component-status-table.html>`__  to view the attributes and status of the installed keyring component when in use.
+
+The explanation of when these are used can be found in
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: when these are used - What does "these" stand for?

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+:ref:`how_xtrabackup_works`.
+
+Creating a User with minimum privileges
++++++++++++++++++++++++++++++++++++++++++++++
+
+An SQL example of creating a database user with the minimum privileges required
+to make full backups would be:
```
Comment (alina-derkach-oaza) [inline storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: would be: -> is the following:

**Review by alina-derkach-oaza — COMMENTED** [review]
See my comments

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 2 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: Done.

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 3 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: Updated.

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 4 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: Rewritten

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 5 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: Rewritten.

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 6 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: Rewritten.

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 7 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: Rewritten.

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 8 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: fixed.

File: storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst (hunk 9 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst]: Rewrote the paragraph.

File: storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst (hunk 10 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst]: Rewritten.

File: storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst (hunk 11 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/configuring.rst]: Rewritten.

File: storage/innobase/xtrabackup/doc/source/prerequisites/connections.rst (hunk 12 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/connections.rst]: Rewritten

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 13 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: fixed.

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 15 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: Fixed.

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 14 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: Rewritten

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 17 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: Updated.

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 18 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: Rewritten.

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 19 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: rewritten

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 20 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: Rewritten

File: storage/innobase/xtrabackup/doc/source/glossary.rst (hunk 1 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/glossary.rst]: Fixed.

File: storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst (hunk 16 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/prerequisites/permissions.rst]: Rewritten

**Review by alina-derkach-oaza — APPROVED** [review]

**Comment by patrickbirch** [comment]
We will redo when we have the ability to preview the ticket

**Pull request closed without merging**